.github/workflows/ci.yml

```diff
@@ -23,6 +23,10 @@ on:
       - main
   workflow_dispatch:
 
+concurrency:
```

Comment:
Should not it be on job level rather than on global?

Reply:
I guess you could do it on jobs, but I'm trying to understand why that would be wanted.  If you push a new commit, there isn't a granular control of the jobs in a workflow, you'd like to see all rerun.  Let me know, I can change it but it seems unnecessary.

Reply:
I'm worried if concurrency limits number of simultaneous jobs in same workflow, Github Workflows docs are ambiguous about that: is one type of job limited or all of them.

Reply:
Ah, so what this will do is group all the jobs run under this workflow together, so there is concurrency for all the jobs in the workflow, what is not allowed is another workflow from this PR named CI to be run.  This concurrency check essentially makes it equivalent of what occurs on Azure Pipelines. Instead of checking job to job, this will check workflow to workflow, so all the jobs in this workflow should run concurrently.